runtime/src/mem/jemalloc/mem-jemalloc.c

```diff
@@ -155,6 +157,11 @@ static inline void* alignHelper(void* base_ptr, size_t offset, size_t alignment)
 // *** Chunk hook replacements *** //
 // See http://www.canonware.com/download/jemalloc/jemalloc-latest/doc/jemalloc.html#arena.i.chunk_hooks
 
+// jemalloc < 4.0 didn't support chunk_hooks_t
+// avoid unused function warnings by only defining these if they could be used
+#if JEMALLOC_VERSION_MAJOR >= 4
```

Comment:
Should check for 4.1 instead of 4.0 -- 4.1 changed `opt.narenas` from `size_t` to `unsigned`so our code won't work with < 4.1

https://github.com/jemalloc/jemalloc/releases/4.1.0 -- Incompatible changes sections

https://github.com/chapel-lang/chapel/commit/c20bdc7f5f7cfe75cfe8999a12e23fdcfde92ce1

We _could_ change our shim to deal with the incompatibly, but realistically I can't imagine that anybody will end up using a version between 4.0 and 4.1.